tests/integration/awslambda/test_lambda.py

```diff
@@ -377,6 +379,31 @@ def test_runtime_introspection_arm(self, lambda_client, create_lambda_function,
         invoke_result = lambda_client.invoke(FunctionName=func_name)
         snapshot.match("invoke_runtime_arm_introspection", invoke_result)
 
+    @pytest.mark.skipif(
+        not use_docker(),
+        reason="Monkeypatching of Docker flags not applicable if run locally",
+    )
```

Comment:
Same as for the other PR, why wouldn't it be applicable?

Reply:
Monkeypatching works but Docker-flags are not used with the local executor in the old lambda provider because no new Docker container is started. Hence, DOCKER flags are not applicable and Docker tests with the local executor fail.

Reply:
I clarified the reason and added a new is_old_local_executor() util. This needs to be extended to the new local executor replacement (i.e., the self-managed workers) because they do not spawn new containers either.